packages/scripts/src/monitor-dvm/common.ts

```diff
@@ -0,0 +1,118 @@
+import { getChainIdByUrl, getLatestBlockNumberByUrl } from "../utils/utils";
+import { VotingV2Ethers } from "@uma/contracts-node";
+import { delay } from "@uma/financial-templates-lib";
+import { BigNumber, utils } from "ethers";
+
+interface BotModes {
+  unstakesEnabled: boolean;
+  stakesEnabled: boolean;
+  governanceEnabled: boolean;
+  deletionEnabled: boolean;
+  emergencyEnabled: boolean;
+  rolledEnabled: boolean;
+  governorTransfersEnabled: boolean;
+  mintsEnabled: boolean;
+}
+
+export interface MonitoringParams {
+  jsonRpcUrl: string;
+  chainId: number;
+  startingBlock: number;
+  endingBlock: number;
+  pollingDelay: number;
+  botModes: BotModes;
+  unstakeThreshold: BigNumber;
+  stakeThreshold: BigNumber;
+  governorTransfersThreshold: BigNumber;
+  mintsThreshold: BigNumber;
+}
+
+export const initMonitoringParams = async (env: NodeJS.ProcessEnv): Promise<MonitoringParams> => {
+  if (!env.CUSTOM_NODE_URL) throw new Error("CUSTOM_NODE_URL must be defined in env");
+  const jsonRpcUrl = env.CUSTOM_NODE_URL;
+
+  const chainId = await getChainIdByUrl(jsonRpcUrl);
+
+  if (!chainId || (chainId != 1 && chainId != 5)) throw new Error("This script should be run on mainnet or goerli");
+
+  // Default to 1 minute polling delay.
+  const pollingDelay = env.POLLING_DELAY ? Number(env.POLLING_DELAY) : 60;
+
+  if (pollingDelay === 0 && (!env.STARTING_BLOCK_NUMBER || !env.ENDING_BLOCK_NUMBER)) {
+    throw new Error("Must provide STARTING_BLOCK_NUMBER and ENDING_BLOCK_NUMBER if running serverless");
+  }
+
+  // If no block numbers are provided, default to the latest block.
+  const latestBlockNumber = await getLatestBlockNumberByUrl(jsonRpcUrl);
+  const startingBlock = env.STARTING_BLOCK_NUMBER ? Number(env.STARTING_BLOCK_NUMBER) : latestBlockNumber;
+  const endingBlock = env.ENDING_BLOCK_NUMBER ? Number(env.ENDING_BLOCK_NUMBER) : latestBlockNumber;
+  if (startingBlock > endingBlock) {
+    throw new Error("STARTING_BLOCK_NUMBER must be less than or equal to ENDING_BLOCK_NUMBER");
+  }
+
+  const botModes = {
+    unstakesEnabled: env.UNSTAKES_ENABLED === "true",
+    stakesEnabled: env.STAKES_ENABLED === "true",
+    governanceEnabled: env.GOVERNANCE_ENABLED === "true",
+    deletionEnabled: env.DELETION_ENABLED === "true",
+    emergencyEnabled: env.EMERGENCY_ENABLED === "true",
+    rolledEnabled: env.ROLLED_ENABLED === "true",
+    governorTransfersEnabled: env.GOVERNOR_TRANSFERS_ENABLED === "true",
+    mintsEnabled: env.MINTS_ENABLED === "true",
+  };
```

Comment:
I could see us wanting to make some of these errors rather than warnings. Have you considered adding a config element to configure the log level? It also might just be obvious which ones should be errors, so configuration may be overkill, but I definitely think some are error-worthy events.

Reply:
agree, for now I just updated hard-coded log levels in b7cd55cd4d84d1c1ed987dc041bbab53d38b02f4. Now emergency proposal, spam deletion, rolled votes, large governor transfers and outsized UMA mints will error.